app/controllers/registrations_controller.rb

```diff
@@ -558,9 +558,24 @@ def stripe_webhook
   end
 
   def payment_completion
-    registration = Registration.includes(:competition).find(params[:id])
+    uses_v2 = params[:attendee_id].present?
+
+
+    registration = if uses_v2
+                     competition_id, user_id_s = params[:attendee_id].split("-")
+                     user_id = user_id_s.to_i
+                     MicroserviceRegistration.find_by(competition_id: competition_id, user_id: user_id)
```

Comment:
I'm sorry to say that the `find_by` entrypoint won't work as expected. It does not communicate with the microservice and will only rely on results that were previously cached.

(Ye, I realize that this is a drawback to my glued-together Microservice+AR integration, and it should probably be improved. But for the time being, we have to accept that we should not use the direct finder methods. See my longer comment above for an alternative suggestion.)

Reply:
well I'm sorry to say that this works already :upside_down_face:, you can check in the old method. I don't mind just using the stripe id instead though

Reply:
It's because the Microservice registration is definitely there because it's created when the payment intent is created